src/raft_server.c

```diff
@@ -1224,6 +1235,8 @@ void raft_offer_log(raft_server_t* me_, raft_entry_t* entries,
         raft_node_t* node = raft_get_node(me_, node_id);
         int is_self = node_id == raft_get_nodeid(me_);
 
+        assert(NULL != node);
```

Comment:
How about moving this assert call to line 1225/1236, before the is_self line?

Actually, I have mixed feelings about this... Due to user data, it seems simpler to ask the log_offer/log_pop implementations to call raft_add_node/raft_remove_node. Otherwise, raft_offer_log needs some way to get user data pointers when calling raft_add_node. (Its current raft_add_node call always pass NULL user data.) Thoughts?